tests/shared/models/selection-state.spec.js

```diff
@@ -129,16 +131,101 @@ describe('Model: selectionState', function () {
         m.chai.expect(drive).to.be.undefined
       })
     })
   })
 
+  describe('given several drives', function () {
+    beforeEach(function () {
+      this.drives = [
+        {
+          device: '/dev/sdb',
+          description: 'DataTraveler 2.0',
+          size: 999999999,
+          mountpoint: '/media/UNTITLED',
+          name: '/dev/sdb',
+          system: false,
+          protected: false
+        },
+        {
+          device: '/dev/disk2',
+          name: 'USB Drive 2',
+          size: 999999999,
+          protected: false
+        },
+        {
+          device: '/dev/disk3',
+          name: 'USB Drive 3',
+          size: 999999999,
+          protected: false
+        }
+      ]
+
+      availableDrives.setDrives(this.drives)
+
+      selectionState.addDrive(this.drives[0].device)
+      selectionState.addDrive(this.drives[1].device)
+    })
+
+    afterEach(function () {
+      selectionState.clear()
+      availableDrives.setDrives([])
+    })
+
+    it('should be able to add more drives', function () {
+      selectionState.addDrive(this.drives[2].device)
+      m.chai.expect(selectionState.getSelectedDevices()).to.deep.equal(_.map(this.drives, 'device'))
+    })
+
+    it('should be able to remove drives', function () {
+      selectionState.removeDrive(this.drives[1].device)
+      m.chai.expect(selectionState.getSelectedDevices()).to.deep.equal([ this.drives[0].device ])
+    })
+
+    it('current drive should be affected by add order', function () {
+      m.chai.expect(selectionState.getCurrentDrive()).to.deep.equal(this.drives[0])
+      selectionState.toggleDrive(this.drives[0].device)
+      selectionState.toggleDrive(this.drives[0].device)
+      m.chai.expect(selectionState.getCurrentDrive()).to.deep.equal(this.drives[1])
+      selectionState.toggleDrive(this.drives[1].device)
+      m.chai.expect(selectionState.getCurrentDrive()).to.deep.equal(this.drives[0])
+    })
+
+    it('should keep system drives selected', function () {
+      const systemDrive = {
+        device: '/dev/disk0',
+        name: 'USB Drive 0',
+        size: 999999999,
+        protected: false,
+        system: true
+      }
+
+      const newDrives = [ ..._.initial(this.drives), systemDrive ]
+      availableDrives.setDrives(newDrives)
+
+      selectionState.addDrive(systemDrive.device)
+      availableDrives.setDrives(newDrives)
+      m.chai.expect(selectionState.getSelectedDevices()).to.deep.equal(_.map(newDrives, 'device'))
+    })
+
+    it('should be able to remove a drive', function () {
+      m.chai.expect(selectionState.getSelectedDevices().length).to.equal(2)
+      selectionState.toggleDrive(this.drives[0].device)
+      m.chai.expect(selectionState.hasDrive()).to.be.true
+      m.chai.expect(selectionState.getSelectedDevices().length).to.equal(1)
+      m.chai.expect(selectionState.getSelectedDevices()).to.deep.equal([ this.drives[1].device ])
+    })
+  })
```

Comment:
Some of these unit-tests feel like a bit of a mish-mash? Could you come up with a neat & logical set of more thorough unit-tests?